packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -3631,6 +3632,62 @@ void main() {
     expect(tester.takeException(), isNull);
   });
 
+  testWidgets('NavigationRail labels shall not overflow if longer texts provided - extended', (WidgetTester tester) async {
+
+    // The navigation rail has a narrow width constraint. The text should wrap.
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(useMaterial3: true),
+      home: Builder(
+        builder: (BuildContext context) {
+          return MediaQuery.withClampedTextScaling(
+            minScaleFactor: 1.0,
+            maxScaleFactor: 1.0,
+            child: Scaffold(
+              body: Row(
+                children: <Widget>[
+                  SizedBox(
+                    width: 140.0,
+                    child: NavigationRail(
+                      selectedIndex: 1,
+                      extended: true,
+                      destinations: const <NavigationRailDestination>[
+                        NavigationRailDestination(
+                          icon: Icon(Icons.favorite_border),
+                          selectedIcon: Icon(Icons.favorite),
+                          label: Text('Abc'),
+                        ),
+                        NavigationRailDestination(
+                          icon: Icon(Icons.bookmark_border),
+                          selectedIcon: Icon(Icons.bookmark),
+                          label: Text('Very long bookmark text for navigation destination'),
+                        ),
+                      ],
+                    ),
+                  ),
+                  const Expanded(
+                    child: Text('body'),
+                  ),
+                ],
+              ),
+            ),
+          );
+        },
+      ),
+    ));
+
+    expect(find.byType(NavigationRail), findsOneWidget);
+    final Finder normalLabelNavDestinationFinder = find.text(
+      'Abc',
+    );
+    final Finder longLabelNavDestinationFinder = find.text(
+      'Very long bookmark text for navigation destination',
+    );
+    expect(normalLabelNavDestinationFinder, findsOneWidget);
+    expect(longLabelNavDestinationFinder, findsOneWidget);
```

Comment:
Maybe consider adding variables for the labels (for instance, `normalLabel` and `longLabel`) and using them for the repro code and for the finders:
```suggestion
    expect(find.text(normalLabel), findsOneWidget);
    expect(find.text(longLabel), findsOneWidget);
```

Reply:
Updated as per the suggestion.